core/dbt/parser/schemas.py

```diff
@@ -905,12 +921,39 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
             if node.patch_path:
                 package_name, existing_file_path = node.patch_path.split("://")
                 raise_duplicate_patch_name(patch, existing_file_path)
+
             source_file.append_patch(patch.yaml_key, unique_id)
             # If this patch has config changes, re-calculate the node config
             # with the patch config
             if patch.config:
                 self.patch_node_config(node, patch)
-            node.patch(patch)
+
+            if node.config.enabled:
```

Comment:
I think we don't need this if/else if we just remove it from disabled and put it in nodes as mentioned in the previous comment.  Just add back the node.patch(patch)